Fix current session and undo/redo bugs

erisjacey opened the pull request:
- Create AttributesList.java for further abstraction
- Refactor names to improve code quality
- Fix bugs wrt current session, specifically with:
   - session commands
   - `enterses` (now commits)
- Add unit testing
   - VersionedStudentList.java
   - VersionedSessionList.java
   - VersionedAttributesList.java

erisjacey committed:
Created AttributesList.java

erisjacey committed:
Refactor names

erisjacey committed:
Fix bugs related to current session

- Current session details get messed up when
session commands are performed

erisjacey committed:
Fix bugs related to current session

- `enterses` now commits

erisjacey committed:
Add unit testing

- VersionedStudentList.java
- VersionedSessionList.java
- VersionedAttributesList.java

erisjacey committed:
Merge branch 'master' into v1.4.1-fix_bugs

marcustw commented on src/main/java/atas/model/session/AttributesList.java:
```diff
@@ -0,0 +1,88 @@
+ */
+public class AttributesList implements ReadOnlyAttributesList {
+
+    protected final ObservableList<Attributes> attributes;
```
Perhaps this could be private-access?

erisjacey replied:
Thanks for the suggestion, will be refactoring some methods to make this possible!

marcustw reviewed (approved):
LGTM!

erisjacey committed:
Refactor methods

- From VersionedAttributesList.java -> AttributesList.java
   - Allows fields in AttributesList.java to be private
